### PR TITLE
Add catalog item

### PR DESCRIPTION
**Pull request by vbauzysvmware**
Added implementation for catalog item recourse management.

Reference: #93

### CODE REVIEW

File: vcd/resource_vcd_catalog_item_test.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+func TestAccVcdCatalogItemBasic(t *testing.T) {
+
+	var catalogItem govcd.CatalogItem
+	var params = StringMap{
+		"Org":             testConfig.VCD.Org,
+		"Vdc":             testConfig.VCD.Vdc,
+		"Catalog":         testConfig.VCD.Catalog.Name,
+		"CatalogItemName": TestAccVcdCatalogItem,
+		"Description":     TestAccVcdCatalogItemDescription,
+		"ovaPath":         "../test-resources/test_vapp_template.ova",
```
Comment (dataclouder) [inline vcd/resource_vcd_catalog_item_test.go]: The field names `ovaPath`, `pieceSize`, and `progress` should be capitalized.
Their values should be included in the `testConfig` structure.
I suggest renaming `progress` to `UploadProgress` and `pieceSize` to `UploadChunkSize`.

Comment (vbauzysvmware) [reply vcd/resource_vcd_catalog_item_test.go]: Thank you. Fixed.

**Comment by hodgesrm** [comment]
I'll review this later today.

File: vcd/resource_vcd_catalog_item_test.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+const testAccCheckVcdCatalogItemBasic = `
+resource "vcd_catalog_item"  "{{.CatalogItemName}}" {
+org = "{{.Org}}"
+vdc = "{{.Vdc}}"
+catalog = "{{.Catalog}}"
+
+name = "{{.CatalogItemName}}"
+description = "{{.Description}}"
+ova_path = "{{.OvaPath}}"
+upload_piece_size = {{.UploadChunkSize}}
```
Comment (lvirbalas) [inline vcd/resource_vcd_catalog_item_test.go]: See previous comment, we should really choose one - piece or chunk - and use that word consistently for the same thing.

**Pull request merged**